test/storage.py

```diff
@@ -9,27 +9,34 @@
 import urllib.parse
 
 import pytest
+import userstorage
+
+
+userstorage.missing_handler = pytest.xfail
 
 
 class Backend:
     """
-    Wrap a userstorage.Backend, adding a url and context manager interface to
-    simplify fixtures.
+    Wrap a userstorage.Backend, adding a url and the can_detect_sector_size
+    attribute.
     """
 
     def __init__(self, storage, can_detect_sector_size=True):
-        if not storage.exists():
-            pytest.xfail("Storage {} is not available".format(storage.name))
-
         self._storage = storage
-        self.path = storage.path
         self.url = urllib.parse.urlparse("file:" + storage.path)
-        self.sector_size = storage.sector_size
         self.can_detect_sector_size = can_detect_sector_size
 
+    @property
+    def path(self):
+        return self._storage.path
+
+    @property
+    def sector_size(self):
+        return self._storage.sector_size
+
     def __enter__(self):
-        self._storage.setup()
+        self._storage.__enter__()
         return self
 
     def __exit__(self, *args):
-        self._storage.teardown()
+        self._storage.__exit__(args)
```

Comment:
Must be:

    self._storage.__exit__(*args)

But it will be better to use the standard signature[1]:

    def __exit__(self, t, v, tb):
        self._storage.__exit__(t, v, tb)

[1] https://docs.python.org/3/reference/datamodel.html#object.__exit__

Reply:
I agree. But I'll just use it the `self._storage.__exit__(*args)` version, since I just remove the context management from the `Backend` class in the next commit (after what we discussed).